### DISCUSSION TITLE
Have a toggle button on page to show another page when clicked and then make it open just a little bit

### DISCUSSION THREAD

**Discussion by henkegithub** [neuefische/questions#685] [Web] [answered]
https://github.com/henkegithub/instroke-app/issues/13

Is this possible? if so any hints and keywords / how tos?

**Comment by Kheirah** [comment]
You essentially want the shopping card component to slide into view from one side but stay on the same page. Is that correct?

**Reply by henkegithub** [reply]
@Kheirah Yes exactly so u can always easily get back to the order list to remove or add stuff from the menu list
By clicking the shopping card (which is link right now) might change that In a button and put a state on the onclick with a true or false state and when it is clicked it opens up a modal that can be changed on x axis via mouse or touch and if you use the full width the shopping cart button will still be overlapping the modal. and you should always be able to get to the order list back when clicking on the order list

**Reply by Kheirah** [reply]
There is some complexity involved in creating a sliding drawer. It's more difficult than creating a separate page specifically for the shopping cart. For example, you will have to consider how to 
- open and close the component
- style the component such that it overlays the currently displayed page
- make the component accessible in that it closes when pressing `ESC` or clicking on the page that's behind the shopping cart

Having said that, it's up to you to decide whether you want a challenge or whether you want to go with the simpler solution of creating a dedicated page. When you need guidance, I [found an article](https://medium.com/@axionoso/step-by-step-guide-to-react-sliding-drawer-e0f8facf3bab) that implements the basic functionality, but it's not using functions :( it's a class-based implementation. You will have to transfer that to the way we write functional components.

**Reply by henkegithub** [reply]
@Kheirah challenge sounds good. I checked the article and I wonder what does class-based implementation work and how to change  class-based implementation to functional components? 

like for example this piece of code:

```js

export default class App extends React.Component {
   state = { drawerOpen: false }
drawerToggleClickHandler = () => {
    this.setState({
      drawerOpen: !this.state.drawerOpen
    })
  }
backdropClickHandler = () => {
    this.setState({
      drawerOpen: false
    })
  }
   render(){
      let backdrop;
      if(this.state.drawerOpen){
        backdrop = <Backdrop close={this.backdropClickHandler}/>;
       }
      return(

         <div>
           < SlideDrawer show={this.state.drawerOpen} />
           { backdrop }
           < MainPage toggle={this.drawerToggleClickHandler} />
         </div>
      )
    }
}

```


```js

export default class App extends React.Component {
   state = { drawerOpen: false }
drawerToggleClickHandler = () => {
    this.setState({
      drawerOpen: !this.state.drawerOpen
    })
  }
backdropClickHandler = () => {
    this.setState({
      drawerOpen: false
    })
  }
   render(){
      let backdrop;
      if(this.state.drawerOpen){
        backdrop = <Backdrop close={this.backdropClickHandler}/>;
       }

```

it would be like:

```js

export default function App () {
drawer,drawerOpen useState (true)}

//how to change the render to a function to work with in our components?

 render(){
      let backdrop;
      if(this.state.drawerOpen){
        backdrop = <Backdrop close={this.backdropClickHandler}/>;
       }

//and I guess the return is just like we use to do it:

 return(

         <div>
           < SlideDrawer show={this.state.drawerOpen} />
           { backdrop }
           < MainPage toggle={this.drawerToggleClickHandler} />
         </div>
      )

```

**Reply by Kheirah** [reply, marked as answer]
This is a perfect task for ChatGPT, actually. You can ask to translate the class component to a functional one and you get:
```js
import React, { useState } from 'react';

const App = () => {
  const [drawerOpen, setDrawerOpen] = useState(false);

  const drawerToggleClickHandler = () => {
    setDrawerOpen(!drawerOpen);
  };

  const backdropClickHandler = () => {
    setDrawerOpen(false);
  };

  return (
    <div>
      <SlideDrawer show={drawerOpen} />
      {drawerOpen && <Backdrop close={backdropClickHandler} />}
      <MainPage toggle={drawerToggleClickHandler} />
    </div>
  );
};

export default App;
```

**Reply by henkegithub** [reply]
@Kheirah Might look into ChatGPT then! Cool stuff! Thanks for the help, ill try it out myself and give an update after I run into more issues!

**Reply by henkegithub** [reply]
I think I got it: https://instroke-app-git-feature-menu-overview-henkegithub.vercel.app/order-list

**Reply by Kheirah** [reply]
Looks amazing! Congrats! 🥳
